addon-test-support/@ember/test-helpers/-utils.ts

```diff
@@ -1,5 +1,43 @@
 /* globals Promise */
-import { Promise as RSVPPromise } from 'rsvp';
+
+import RSVP from 'rsvp';
+import Ember from 'ember';
+import { run } from '@ember/runloop';
+import hasEmberVersion from './has-ember-version';
+
+const backburner = (run as any).backburner;
+
+export class _Promise extends RSVP.Promise<void> {}
+
+RSVP.configure('async', (callback, promise) => {
+  if (promise instanceof _Promise) {
+    if (typeof Promise !== 'undefined') {
+      // use real native promise semantics whenever possible
+      Promise.resolve().then(() => callback(promise));
+    }  else {
+      // fallback to using RSVP's natural `asap` (**not** the fake
+      // one configured by Ember...)
+      RSVP.asap(callback, promise);
+    }
+  } else {
+    // this is the default behaviors from Ember:
+    //   https://github.com/emberjs/ember.js/blob/v3.4.6/packages/ember-testing/lib/ext/rsvp.js
```

Comment:
does this mean we're copying stuff out of Ember? What if that code changes in the future and gets out of sync?

Reply:
@rwjblue can we not just copy the existing config async out instead of copying it?

Reply:
Yep, great idea. Will update (along with some additional comments explaining the overall issue).